mcnp/meshtal_funcs.h

```diff
@@ -22,27 +22,12 @@ extern "C" {
  * Macro to access symbol of fortran function 'func' in module 'mod'
  */
 #ifndef FORT_FUNC
-#if defined(__clang__)
-/* clang/gfortran naming scheme" ___<module>_mod_MOD_<function> */
-#define FORT_FUNC( mod, func ) __##mod##_MOD_##func
-#elif defined __INTEL_COMPILER
-/* intel fortran: name mangling is '<module>_mp_<function>_' */
+#if defined INTEL
+// intel fortran: name mangling is '<module>_mp_<function>_'
 #define FORT_FUNC( mod, func ) mod##_mp_##func##_
-#elif defined(__CNUC__) || defined(__GNUG__)
-#if ( __GNUC__ > 4 ) || ( __GNUC__ == 4  && __GNUC_MINOR__ >= 3 )
-/* gcc/gfortran 4.3 and above: name mangling is '__<module>_MOD_<function>' */
-#define FORT_FUNC( mod, func ) __##mod##_MOD_##func
-#elif ( __GNUC__ == 4)
-/* gcc/gfortran < 4.3: name mangling is '__<module>__<function>' */
-#define FORT_FUNC( mod, func ) __##mod##__##func
 #else
-#error "Some unknown GCC compiler version"
-#endif
-#else
-#error "Some unknown C++ compiler"
-/* Comment out this error to force compile to proceed; it may or may not work */
-#error "DagMC: unknown compiler with unknown fortran name mangling scheme."
```

Comment:
Also, just a FYI, the reason I'm setting these as preprocessor directives and not using, say, `__INTEL_COMPILER` or `__GNUC__` is that those variables are set based on the C++ compiler, not the fortran compiler because they appear in a C++ file. Since the name mangling in this case depends entirely on the choice of fortran compiler, this seemed like the best way to do it. This wouldn't matter at all if you used gcc/gfortran or icc/ifort, but it becomes a problem if you want to use something like gcc/ifort.